webapp/src/shared/components/Game/useGameControls.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { GamePaddle } from './types';
+import { PADDLE_SLIDE_SPEED } from './constants';
+
+const useGameControls = (paddleRef: React.MutableRefObject<GamePaddle>) => {
+  const dragRef = React.useRef<number>(0);
+
+  const movePaddle = React.useCallback(
+    (e: KeyboardEvent) => {
+      const key = e.key;
+      const paddle = paddleRef.current;
+
+      if (key === 'ArrowRight') {
+        paddleRef.current = {
+          ...paddle,
+          slide: PADDLE_SLIDE_SPEED,
+        };
+      } else if (key === 'ArrowLeft') {
+        paddleRef.current = {
+          ...paddle,
+          slide: -1 * PADDLE_SLIDE_SPEED,
+        };
+      }
+    },
+    [paddleRef],
+  );
+
+  const dragPaddle = React.useCallback(
+    (e: TouchEvent) => {
+      const paddle = paddleRef.current;
+      const dragPos = dragRef.current;
+      dragRef.current = e.touches[0].clientX;
+
+      if (dragPos)
+        paddleRef.current = {
+          ...paddle,
+          x: paddle.x - (dragPos - dragRef.current),
+        };
+    },
+    [paddleRef],
+  );
+
+  const stopPaddle = React.useCallback(
+    (e: KeyboardEvent) => {
+      const key = e.key;
+      const paddle = paddleRef.current;
+
+      if (key === 'ArrowRight' || key === 'ArrowLeft') {
+        paddleRef.current = {
+          ...paddle,
+          slide: 0,
+        };
+      }
+    },
+    [paddleRef],
+  );
+
+  window.addEventListener('keydown', movePaddle, false);
+  window.addEventListener('keyup', stopPaddle, false);
+  window.addEventListener('touchmove', dragPaddle, false);
+  window.addEventListener('contextmenu', (e) => e.preventDefault());
```

Comment:
issue: This could potentially memory leak if we don't remove the event listeners

suggestion: We could place the `window.addEventListener()` calls in a `useEffect` and call `window.removeEventListener()` on cleanup

Here is an example: https://beta.reactjs.org/reference/react/useEffect#listening-to-a-global-browser-event

Reply:
not sure if that would cause a memory leak but a registered listener which we are not going to use so, agree and removed 👍